Formula/php.rb

```diff
@@ -17,26 +17,32 @@ class Php < Formula
   depends_on "argon2"
   depends_on "aspell"
   depends_on "autoconf"
-  depends_on "curl" if MacOS.version < :lion
+  depends_on "brotli"
   depends_on "freetds"
   depends_on "freetype"
   depends_on "gettext"
   depends_on "glib"
   depends_on "gmp"
   depends_on "icu4c"
   depends_on "jpeg"
-  depends_on "libiconv" if DevelopmentTools.clang_build_version >= 1000
   depends_on "libpng"
   depends_on "libpq"
   depends_on "libsodium"
   depends_on "libzip"
-  depends_on "openldap" if DevelopmentTools.clang_build_version >= 1000
+  depends_on "nghttp2"
+  depends_on "openldap"
   depends_on "openssl"
   depends_on "pcre"
   depends_on "sqlite"
   depends_on "unixodbc"
   depends_on "webp"
 
+  # avoids mixed crypto and ldap
```

Comment:
We were discussing to create a `curl-openssl` Formula instead of resourcing it especially since multiple versions of PHP will use it. We would want to avoid managing the same resource multiple times if it can be avoided. I was starting to look at this but never got to a point where I was happy but I was imagining a formula that's completely dependent on other homebrew formula for its dependencies. Perhaps even resourcing nghttp2 in it to get around the boost dependency to be able to support http2 but I'm not sure what the other maintainers think of that.

Reply:
I agree that having a new formula would be a more ideal solution. If there was some kind of online discussion, it would be nice to link to it so we can refer to it later when people land here or on the linked issue.